File: C++/Computer.cpp
#include "Computer.h"

Computer::Computer(OthelloPtr othello) : Player(othello)
{
}

// simple greedy player
bool Computer::onTurn()
{
	const Board& board = othello->getBoard();

	int max = 0, tx, ty;
	for (int y = 0; y < board.height; ++y)
		for (int x = 0; x < board.width; ++x) {
			int c = board.getCount(x, y);
			if (c > max || ((c == max) && rand() % 2)) {
				max = c;
				tx = x; ty = y;
			}
		}

	othello->putStone(tx, ty);

	return Player::onTurn();
}

File: Python/computer.py
from player import *
from board import *
from random import randint

class Computer(Player):
    def __init__(self, othello):
        super().__init__(othello)

    def onTurn(self):
        board = self.othello.board

        max = 0
        for y in range(board.height):
            for x in range(board.width):
                c = board.count[y][x]

                if c > max or (c == max and randint(0, 1) == 0):
                    max, tx, ty = c, x, y

        self.othello.putStone(tx, ty)
